### PR TITLE
CLIO-36187: thFilter wrapper directive

### PR DESCRIPTION
**Pull request by dtomalty**

### CODE REVIEW

File: themis_components/thFilter/filters/input/thFilter.input.directive.coffee [hunk 1]
```diff
@@ -26,3 +24,7 @@ angular.module 'ThemisComponents'
         @filter.clearValue()
 
       return
+    compile: ->
```
Comment (dtomalty) [inline themis_components/thFilter/filters/input/thFilter.input.directive.coffee]: These were moved into the compile step (here and in thFilter.select.directive) to accommodate using options from the `th-filter` wrapper directive. This is the first time we can access the parent controller.

File: themis_components/thFilter/thCustomFilters.directive.coffee [hunk 2]
```diff
@@ -1,48 +1,21 @@
...
 .directive "thCustomFilters", ($http, CustomFilterConverter) ->
   restrict: "E"
+  require: "?^^thFilter"
   scope:
     options: "="
   bindToController: true
   controllerAs: "thCustomFilters"
   template: require "./thCustomFilters.template.html"
   controller: ->
-    {
```
Comment (dtomalty) [inline themis_components/thFilter/thCustomFilters.directive.coffee]: Moved into compile so we can access the parent controller.

File: themis_components/thFilter/filters/input/thFilter.input.template.html [hunk 3]
```diff
@@ -3,6 +3,7 @@
     ng-model="thFilterInput.filter.model"
     ng-attr-placeholder="{{thFilterInput.filter.placeholder || 'Enter filter term…'}}"
     ng-keypress="thFilterInput.onKeypress($event)"
+    ng-blur="thFilterInput.ngBlur()"
```
Comment (craigcarlyle) [inline themis_components/thFilter/filters/input/thFilter.input.template.html]: This function's name should be consistent with the keypress function. Either `ngKeypress` or `onBlur`. No preference, but it should be the same style.

File: themis_components/thFilter/index.scss [hunk 4]
```diff
@@ -1,6 +1,31 @@
...
+      align-self: flex-end;
+    }
+  }
+}
+
 .th-static-filters,
 .th-custom-filters,
 .th-search-row {
+  label.th-label .label-text {
+    color: black;
```
Comment (craigcarlyle) [inline themis_components/thFilter/index.scss]: Can we change this to `#000` pleeaaase?

File: themis_components/thFilter/thFilter.template.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<div class="th-filter">
+  <div class="th-filter-header">
+    <div class="filter-header-text">Filters</div>
+    <th-button
+      class="clear-filters"
+      ng-click="thFilter.clearFilters()"
+      type="standard"
```
Comment (craigcarlyle) [inline themis_components/thFilter/thFilter.template.html]: `type="standard"` may have to change with the newest changes to thButton.

Comment (dtomalty) [inline themis_components/thFilter/thFilter.template.html]: GOOD EYE.

File: themis_components/thFilter/index.scss [hunk 6]
```diff
@@ -1,6 +1,31 @@
+.th-filter {
+  background: #e8e8e8;
```
Comment (seanhealy) [inline themis_components/thFilter/index.scss]: We should follow the box model closer for rule order here: http://codeguide.co/#css-declaration-order

Comment (seanhealy) [inline themis_components/thFilter/index.scss]: Also, I assume you consulted design on this part of it?

File: themis_components/thFilter/thCustomFilters.directive.coffee [hunk 7]
```diff
@@ -51,3 +24,35 @@ angular.module "ThemisComponents"
       @customFilterRows.splice(index, 1)
 
     return
+  compile: ->
+    pre: (scope, element, attrs, controller) ->
```
Comment (seanhealy) [inline themis_components/thFilter/thCustomFilters.directive.coffee]: I assume `controller` here references the `thFilter`'s controller? Can we name it a bit better?

**Pull request merged**